pkg/clusterapi/aws/actuator.go

```diff
@@ -367,35 +339,114 @@ func (a *Actuator) DeleteMachine(machine *clusterv1.Machine) error {
 		return fmt.Errorf("error getting EC2 client: %v", err)
 	}
 
-	terminateInstancesRequest := &ec2.TerminateInstancesInput{
-		InstanceIds: []*string{
-			aws.String(instanceID),
-		},
+	instances, err := a.getRunningInstances(machine, client)
+	if err != nil {
+		return err
+	}
+	if len(instances) == 0 {
+		mLog.Warnf("no instances found to delete for machine")
+		return nil
 	}
-	terminateInstancesResult, err := client.TerminateInstances(terminateInstancesRequest)
+
+	instanceIDs := []*string{}
+	for _, instance := range instances {
+		mLog.WithField("instanceID", *instance.InstanceId).Infof("deleting AWS instance")
```

Comment:
Remove. You're logging the same thing inside the terminateInstances function that gets called right after this.

Reply:
Removed.